libmachine/util/util.go

```diff
@@ -16,14 +15,17 @@ func FindEnvAny(names ...string) string {
 	return ""
 }
 
-func GetProxyURL() string {
-	urlRaw := FindEnvAny("HTTP_PROXY", "HTTPS_PROXY")
-	if urlRaw == "" {
-		log.Debug("env var HTTP_PROXY or HTTPS_PROXY is not found")
-		return ""
+func GetProxyHostnamePortForHost(hostname string) (string, error) {
+	req, err := http.NewRequest("GET", "http://"+hostname, nil)
```

Comment:
I don't think we can put `http://` here because the `ProxyFromEnvironment` will produce a different answer depending on the protocol. Is it possible that `hostname` will have a protocol already?

Reply:
The function is updated to take a parameter named `scheme` for the preferred protocol. 
One note here is that the  HTTP_PROXY and HTTPS_PROXY are set to the same value in the rancher container in the HA installation.